internal/generator/vector/metrics.go

```diff
@@ -1,13 +1,28 @@
 package vector
 
+import (
+	"golang.org/x/sys/unix"
+)
+
 const (
 	InternalMetricsSourceName = "internal_metrics"
 	PrometheusOutputSinkName  = "prometheus_output"
-	PrometheusExporterAddress = "[::]:24231"
 
 	AddNodenameToMetricTransformName = "add_nodename_to_metric"
 )
 
+var PrometheusExporterAddress string
+
+func init() {
+	if fd, err := unix.Socket(unix.AF_INET6, unix.SOCK_STREAM, unix.IPPROTO_IP); err != nil {
```

Comment:
1. Nope
```
$ grep address ~/syslog/vector_ip.toml | grep -v '^#'
address = ":24231"
$ target/debug/vector -c ~/syslog/vector_ip.toml
...
2023-10-09T13:59:02.169084Z  INFO vector::app: Loading configs. paths=["/home/syedriko/syslog/vector_ip.toml"]
2023-10-09T13:59:02.170997Z ERROR vector::cli: Configuration error. error=invalid socket address syntax
in `sinks.prometheus_output`
$ 
```

2.  ...would now fail on non-IPv6 enabled hosts
We don't have any, not even in QA, otherwise we would've caught this issue earlier. All the CI tests pass.
I like this approach exactly because it's unavoidable and unconditional. We can tell an IPv4-only host if we ever have it automated, in the tests and expect the correct config gen output accordingly.

Reply:
@syedriko Yes, this is nothing that "will happen immediately". Having the code in `init()` that changes the value of the variable before it is used the first might just be _surprising_ if you look on the code and don't know about this yet (or have forgotten it in a few months), so this is a construct I'd try to avoid.

But it's not _wrong_, so consider my feedback on this optional.

Reply:
PrometheusExporterAddress used to be a const, what I'm doing is the best approximation to a const I could come up with.